src/capped_hashmap.rs

```diff
@@ -0,0 +1,172 @@
+use std::{
+    cmp::Eq,
+    collections::{HashMap, VecDeque},
+    hash::Hash,
+};
+
+pub struct CappedHashMap<K, V>
+where
+    K: Hash + Eq + Copy + Clone,
+{
+    capacity: usize,
```

Comment:
imo this field is unnecessary as you can just get it from `self.inner.capacity()`